src/workflows-app/FeaturedWorkflows.ts

```diff
@@ -0,0 +1,250 @@
+import _ from 'lodash/fp';
+import { Dispatch, Fragment, SetStateAction, useCallback, useState } from 'react';
+import { div, h, h2 } from 'react-hyperscript-helpers';
+import { AnalysesData } from 'src/analysis/Analyses';
+import { getCurrentApp, getIsAppBusy } from 'src/analysis/utils/app-utils';
+import { appToolLabels } from 'src/analysis/utils/tool-utils';
+import { Clickable } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import TooltipTrigger from 'src/components/TooltipTrigger';
+import { Cbas } from 'src/libs/ajax/workflows-app/Cbas';
+import colors from 'src/libs/colors';
+import { notify } from 'src/libs/notifications';
+import { useCancellation, useOnMount, usePollingEffect } from 'src/libs/react-utils';
+import { AppProxyUrlStatus, workflowsAppStore } from 'src/libs/state';
+import { withBusyState } from 'src/libs/utils';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+import { ImportWorkflowModal } from 'src/workflows-app/components/ImportWorkflowModal';
+import { WorkflowCard, WorkflowMethod } from 'src/workflows-app/components/WorkflowCard';
+import { FeaturedWorkflow, featuredWorkflowsData } from 'src/workflows-app/fixtures/featured-workflows';
+import { doesAppProxyUrlExist, loadAppUrls } from 'src/workflows-app/utils/app-utils';
+import { CbasPollInterval } from 'src/workflows-app/utils/submission-utils';
+
+type FeaturedWorkflowsProps = {
+  name: string;
+  namespace: string;
+  workspace: WorkspaceWrapper;
+  analysesData: AnalysesData;
+  setSelectedSubHeader: Dispatch<SetStateAction<string>>;
+};
+
+export const FeaturedWorkflows = ({
+  name,
+  namespace,
+  workspace,
+  workspace: {
+    workspace: { workspaceId },
+  },
+  analysesData: { apps, refreshApps },
+  setSelectedSubHeader,
+}: FeaturedWorkflowsProps) => {
+  const [methodsData, setMethodsData] = useState<WorkflowMethod[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  const [methodName, setMethodName] = useState('');
+  const [methodId, setMethodId] = useState('');
+
+  const [importWorkflowModal, setImportWorkflowModal] = useState(false);
+  const [importLoading, setImportLoading] = useState(false);
+  const [successfulImport, setSuccessfulImport] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const signal = useCancellation();
+  const cbasReady = doesAppProxyUrlExist(workspaceId, 'cbasProxyUrlState');
+  const currentApp = getCurrentApp(appToolLabels.CROMWELL, apps);
+
+  const loadRunsData = useCallback(
+    async (cbasProxyUrlDetails) => {
```

Comment:
I wonder if pulling this out into a function defined at the top level and get rid of that "cognitive complexity" code smell... but it doesn't seem that bad to me, personally.